File: server/src/main/java/com/cyberbot/bomberman/server/ClientController.java
package com.cyberbot.bomberman.server;

import com.cyberbot.bomberman.core.models.net.packets.ClientRegisterRequest;
import com.cyberbot.bomberman.core.models.net.packets.ClientRegisterResponse;

public interface ClientController {
    ClientRegisterResponse onClientRegister(ClientRegisterRequest request, ClientControlService service);
}

File: server/src/main/java/com/cyberbot/bomberman/server/ClientController.kt
package com.cyberbot.bomberman.server

import com.cyberbot.bomberman.core.models.net.packets.*

interface ClientController {
    fun onClientRegister(request: ClientRegisterRequest, service: ClientControlService): ClientRegisterResponse?

    fun onLobbyCreate(request: LobbyCreateRequest, client: Client): LobbyCreateResponse?

    fun onLobbyJoin(request: LobbyJoinRequest, client: Client): LobbyJoinResponse?

    fun onGameStart(request: GameStartRequest, client: Client)
}
